backend/auth/handlers.go

```diff
@@ -0,0 +1,340 @@
+		return
+	}
+	if user == nil {
+		c.JSON(http.StatusNotFound, gin.H{"error": errorUsernameNotFound})
+		return
+	}
+	user.Password = ""
+	c.JSON(http.StatusOK, user)
+}
+
+func fetchUser(username string) (*configmodels.User, error) {
+	filter := bson.M{"username": username}
+	rawUser, err := dbadapter.WebuiDBClient.RestfulAPIGetOne(UserAccountDataColl, filter)
+	if err != nil {
+		logger.DbLog.Errorln(err.Error())
+		return nil, err
+	}
+	if len(rawUser) == 0 {
+		return nil, nil
+	}
+	var user configmodels.User
+	err = json.Unmarshal(configmodels.MapToByte(rawUser), &user)
+	if err != nil {
+		logger.AuthLog.Errorln(err.Error())
+		return nil, err
+	}
+	return &user, nil
+}
+
+func PostUserAccount(c *gin.Context) {
+	logger.WebUILog.Infoln("create user account")
+	var user configmodels.User
+	err := c.ShouldBindJSON(&user)
+	if err != nil {
+		logger.AuthLog.Errorln(err.Error())
+		c.JSON(http.StatusBadRequest, gin.H{"error": errorInvalidDataProvided})
+		return
+	}
+	if user.Username == "" {
+		c.JSON(http.StatusBadRequest, gin.H{"error": errorMissingUsername})
+		return
+	}
+	shouldGeneratePassword := user.Password == ""
+	if shouldGeneratePassword {
+		generatedPassword, passwordErr := generatePassword()
+		if passwordErr != nil {
+			logger.AuthLog.Errorln(passwordErr.Error())
+			c.JSON(http.StatusInternalServerError, gin.H{"error": errorCreateUserAccount})
+			return
+		}
+		user.Password = generatedPassword
+	}
+	if !validatePassword(user.Password) {
+		logger.AuthLog.Errorln("invalid password provided")
+		c.JSON(http.StatusBadRequest, gin.H{"error": errorInvalidPassword})
+		return
+	}
+	user.Permissions = USER_ACCOUNT
+	isFirstAccountIssued, err := IsFirstAccountIssued()
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": errorRetrieveUserAccounts})
+		return
+	}
+	if !isFirstAccountIssued {
+		user.Permissions = ADMIN_ACCOUNT
+	}
+	password := user.Password
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(user.Password), bcrypt.DefaultCost)
+	if err != nil {
+		logger.AuthLog.Errorln(err.Error())
+		c.JSON(http.StatusInternalServerError, gin.H{"error": errorCreateUserAccount})
+		return
+	}
+	user.Password = string(hashedPassword)
+	filter := bson.M{"username": user.Username}
+	err = dbadapter.WebuiDBClient.RestfulAPIPostMany(UserAccountDataColl, filter, []interface{}{configmodels.ToBsonM(user)})
+	if err != nil {
+		if strings.Contains(err.Error(), "E11000") {
+			logger.DbLog.Errorln("Duplicate username found:", err)
+			c.JSON(http.StatusBadRequest, gin.H{"error": "user account already exists"})
+			return
+		}
+		logger.DbLog.Errorln(err.Error())
+		c.JSON(http.StatusInternalServerError, gin.H{"error": errorCreateUserAccount})
+		return
+	}
+	if shouldGeneratePassword {
+		c.JSON(http.StatusCreated, gin.H{"password": password})
+		return
+	}
+	c.JSON(http.StatusCreated, gin.H{})
+}
+
+func DeleteUserAccount(c *gin.Context) {
```

Comment:
Do user endpoints belong under `auth`? Probably not, users are objects on their own, independent from the login endpoint,